### PR TITLE
Replacing deprecated packages

### PR DESCRIPTION
**Pull request by npiyush97**
## Hey, I just made a Pull Request!
wip #20593 

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
Hi @npiyush97, can you please update the title to be a bit more descriptive? This makes it hard for people to review. If this is work in progress as you've indicated please leave this as a Draft. 

I do think at this time it's probably too early for this specific change though and it might make more sense to focus on getting everything migrated to Material UI v5

**Comment by npiyush97** [comment]
what if I migrate the whole opencost plugin to v5? @awanlin  @Rugvip

**Comment by awanlin** [comment]
> what if I migrate the whole opencost plugin to v5? @awanlin @Rugvip

That works for me, it will need to be done eventually. What do you think @Rugvip?

**Review by tudi2d — CHANGES_REQUESTED** [review]
Thanks! I believe this should work, but it would be good to make clear that people have to use the `UnifiedThemeProvider` for this to work. It is also one of the first plugins with a MUI v4/v5 mixture, which would be interesting to watch. The goal should really be to migrate the whole plugin to MUI v5 in the future :) Requesting changes just to clarify the usage of the date library :)

slight ping to @mattray as the plugin owner to be aware of the changes.

File: plugins/opencost/package.json [hunk 1]
```diff
@@ -30,8 +30,8 @@
     "@material-ui/core": "^4.9.13",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "^4.0.0-alpha.60",
-    "@material-ui/pickers": "^3.3.10",
     "@material-ui/styles": "^4.11.5",
+    "@mui/x-date-pickers": "6.17.0",
```
Comment (tudi2d) [inline plugins/opencost/package.json:34]: sorry for the addon, but please commit the `yarn.lock` changes as well :)

**Review by tudi2d — CHANGES_REQUESTED** [review]

File: plugins/opencost/src/components/SelectWindow.jsx [hunk 2]
```diff
@@ -16,12 +16,9 @@
 
 import React, { useEffect, useState } from 'react';
 import { makeStyles } from '@material-ui/styles';
-import {
-  MuiPickersUtilsProvider,
-  KeyboardDatePicker,
-} from '@material-ui/pickers';
+import { DatePicker, LocalizationProvider } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
```
[the remark this answers is not in the document]
Comment (npiyush97) [reply plugins/opencost/src/components/SelectWindow.jsx:20]: I had issues with dayjs and luxon.there are some inconsistency with adapterluxon so i used dataadapter.

File: plugins/opencost/package.json (hunk 1 above)
Comment (npiyush97) [reply plugins/opencost/package.json:34]: Will do this weekend

**Comment by npiyush97** [comment]
@mattray yes its working fine but as backstage migrating to v5 might as well update the whole plugin.

**Comment by mattray** [comment]
@npiyush97 I'm unlikely to be able to test until post KubeCon, but I'm OK with the changes in principle

**Comment by tudi2d** [comment]
@npiyush97 up for bringing this over the finish line? any help needed? :)

**Pull request closed without merging**